subsys: tracing: Add macros for tracking list operations

rruuaanng opened the pull request:
Add a set of 'SYS_PORT_TRACK_K_*' wrapper macros for visible access to the global tracking list, and a macro 'SYS_PORT_TRACK_FOREACH' for iterating over the tracking list.

ubieda commented:
@rruuaanng Can you look at the CI failure?

rruuaanng commented:
@ubieda It looks like it's not an issue, the warning is caused by my syntax.
```
#define SYS_PORT_TRACK_FOREACH(object, var)		\
	for (struct object *var = _track_list_##object	\
               ; var != NULL; var++)
```
```
BRACES

include/zephyr/tracing/tracking.h:67  braces {} are required around if/while/for/else
```

ubieda commented on doc/releases/release-notes-4.2.rst:
Keep in mind we'll have to move this to 4.3.0, since 4.2.0 is about to be released now.

kartben replied:
Prob better to even make it a request for change then (although I understand your intention was to show your approval in principle) to avoid fly-by merge when merge window re-opens

ubieda commented on doc/releases/release-notes-4.2.rst:
Keep in mind we'll have to move this to 4.3.0, since 4.2.0 is about to be released now.

ubieda commented:
Would you mind rebasing to resolve conflicts?

edersondisouza reviewed (dismissed):
Just curiosity - do you have use cases for accessing these lists from inside Zephyr? I've imagined they would be used by external tools only, so they would only care about the symbol name.

ubieda commented:
@rruuaanng would you mind rebasing?

rruuaanng commented:
@ubieda Done :)

jhedberg commented on doc/releases/release-notes-4.3.rst:
```diff
@@ -334,6 +348,7 @@ New APIs and options
 
 .. zephyr-keep-sorted-stop
 
+
```
minor: seems like an unrelated change

nashif commented on include/zephyr/tracing/tracking.h:
```diff
@@ -73,6 +73,81 @@ extern struct k_event *_track_list_k_event;
  */
 #define SYS_PORT_TRACK_NEXT(list)((list)->_obj_track_next)
 
+/**
```
i see macros being added, but I do not see them being used, can you please elaborate on the usage of thos macros and why we need them?

rruuaanng committed:
subsystem: tracing: Add macros for tracking list operations

Add a set of `SYS_PORT_TRACK_K_*` wrapper macros for
visible access to the global tracking list, and a macro
`SYS_PORT_TRACK_FOR_EACH` for iterating over the tracking list.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng commented:
@nashif @kartben @jhedberg Thanks for your review :)
